### PR TITLE
Introduce POSIX_DATETIME primitive type represented by int64.

### PR DESCRIPTION
**Pull request by yuslepukhin**

### CODE REVIEW

**Comment by gramalingam** [comment]
LGTM. It would be good to have some toy unit test case for the added code. But, off the top of my head, I am not sure whether we can do anything simple without adding an op or two that uses "datetime". Can we create some dummy op that uses datetime, but is not part of the standard onnx opset?

**Comment by yuslepukhin** [comment]
> 
> 
> LGTM. It would be good to have some toy unit test case for the added code. But, off the top of my head, I am not sure whether we can do anything simple without adding an op or two that uses "datetime". Can we create some dummy op that uses datetime, but is not part of the standard onnx opset?

It looks like ops that are not defined in ONNX are ignored: https://github.com/onnx/onnx/blob/master/onnx/backend/test/case/node/__init__.py#L76

**Comment by gramalingam** [comment]
@houseroad : any feedback on this?

File: onnx/mapping.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
     int(TensorProto.COMPLEX64): np.dtype('complex64'),
     int(TensorProto.COMPLEX128): np.dtype('complex128'),
     int(TensorProto.UINT32): np.dtype('uint32'),
-    int(TensorProto.UINT64): np.dtype('uint64'),
+    int(TensorProto.POSIX_DATETIME): np.dtype('uint64'),
```
Comment (houseroad) [inline onnx/mapping.py]: why remove UNIT64 case?

File: onnx/onnx-ml.proto [hunk 2]
```diff
@@ -592,8 +596,4 @@ message OperatorSetIdProto {
   // The version of the operator set being identified.
   // This field MUST be present in this version of the IR.
   optional int64 version = 2;
-}
-
-// For using protobuf-lite
-option optimize_for = LITE_RUNTIME;
```
Comment (houseroad) [inline onnx/onnx-ml.proto]: Why remove this?

File: onnx/onnx-ml.proto [hunk 3]
```diff
@@ -592,8 +596,4 @@ message OperatorSetIdProto {
...
-
+}
```
Comment (houseroad) [inline onnx/onnx-ml.proto]: endline charactor?

File: onnx/onnxifi.h [hunk 4]
```diff
@@ -38,13 +38,13 @@ extern "C" {
 #endif
```
Comment (houseroad) [inline onnx/onnxifi.h]: please don't mix too much formatting in one PR.

File: onnx/mapping.py (hunk 1 above)
Comment (yuslepukhin) [reply onnx/mapping.py]: Oversight

---
In reply to: [344909641](https://github.com/onnx/onnx/pull/2437#discussion_r344909641) [](ancestors = 344909641)

File: onnx/onnx-ml.proto (hunk 3 above)
Comment (yuslepukhin) [reply onnx/onnx-ml.proto]: This was copied as one of the build products. I will re-do change manually.

---
In reply to: [344909908](https://github.com/onnx/onnx/pull/2437#discussion_r344909908) [](ancestors = 344909908)

File: onnx/onnxifi.h (hunk 4 above)
Comment (yuslepukhin) [reply onnx/onnxifi.h]: Will do the change outside VS which often applies formatting automatically

---
In reply to: [344910141](https://github.com/onnx/onnx/pull/2437#discussion_r344910141) [](ancestors = 344910141)

**Comment by yuslepukhin** [comment]
Build fails bc Conda protobuf package has DLL in it not a static library and therefore it can not link.

**Comment by yuslepukhin** [comment]
@houseroad Can you, please, check if the changes are addressed?

**Comment by gramalingam** [comment]
@houseroad : do the updates look fine to you? Thanks!

**Pull request closed without merging**